src/Microsoft.Extensions.Configuration.AzureAppConfiguration/ConfigurationClientManager.cs

```diff
@@ -103,5 +104,17 @@ public bool UpdateSyncToken(Uri endpoint, string syncToken)
 
             return false;
         }
+
+        public Uri GetEndpointForClient(ConfigurationClient client)
+        {
+            ConfigurationClientWrapper currentClient = _clients.FirstOrDefault(c => c.Client == client);
+            
+            if (currentClient == null)
+            {
+                throw new ArgumentNullException(nameof(client));
+            }
```

Comment:
ArgumentNullException should be thrown if the passed argument is null. In this case, we should check if `client` is null and throw ArgumentNullException. This argument check should be done _before_ we use `client` to find current client.